### PR TITLE
Dwarves finally coming to the station, this was literally five hours …

### PR DESCRIPTION
**Pull request by AnthonyJNgo**
…straight of coding and debugging.
## Changelog
:cl:
add: Dwarves are now a ghost role! They're a bit basic right now, updates coming soon.
del: Beach, Seed Vault, Ash Walkers Syndicate Base, Free Golems, Vet Hospital are all gone now.
tweak: Dwarves is highly experimental, right now they're infinite spawns so if you die due to bugs you can just pop back in.
/:cl:

About The Pull Request

Alright, I wagered a resignation over this PR, so I hope you all understand if it isn't as polished as I wanted it to be. I had THREE different coders delay things on me, AAAAAAAAAAAAAAAAAA. In any case, I worked through the night (started at 11/12, it's 5AM now) and I'm happy to say that we have an extremely basic, but functional version of dwarves. I'll do a tl;dr near the end of this description as well. Basically all ghost roles were removed, and dwarves are replacing them. As it is now I believe dwarves have the same amount of content the other ghost roles had which is very little. Beach bum was just running around on the beach, the plant people just grew plants, and the syndicate roles were honestly garbage. I don't plan on stopping here though, we'll have more things coming, I just wanted literally anything rolled out before the new year came in.

Anyways, here's the tl;dr:

**Additions:**

A part of a snowy Dwarven fortress will now spawn in Lavaland! This is hardcoded to spawn every time, so dwarves will ALWAYS be an available ghost role.

The Altar of Armok is blessed, you can respawn an infinite number of times! This will definitely change, however, for now it can stay like this since I foresee plenty of people dying due to unintended bugs.

Some fluff messages were edited slightly, you most likely won't notice the change.

**Removals:**

All other ghost roles were removed. There is still some golem code and I also left the pirates since those are more of an event than anything.

The ruins associated with those roles were also deleted.

Lavaland atmos has been somewhat gutted. I'll tinker with it more later, but for now it doesn't matter, it's still hostile to humans and it doesn't really affect dwarves.

**Known Issues:**

Dwarves should really either properly breathe CO2, or just have something that binds them to the Z level of their fortress preventing them from raiding the station. As of now they're nobreathe.

Pressure might still harm and kill you as a Dwarf, even though you have the resist traits. I'm looking into this, I'm probably just forgetting something super minor. In any case your starting area is completely habitable and well lit.

The role really shouldn't have infinite uses, but it's fine for now.

There is no way to stop Dwarves from going on station at this point, except for the hazardous Lavaland. Please refrain from doing so, although you wouldn't be able to do anything anyways. Any dwarves seen on the station may be subject to adminbus or punishment.

Obviously the lack of features, but this project is ongoing and will continue to be fleshed out. As it is, I believe it is on par with the removed roles.

DwarfFortress

### CODE REVIEW

File: hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 1]
```diff
@@ -7,6 +7,7 @@
 	name = "Dwarf"
 	id = "dwarf"
 	species_traits = list(EYECOLOR,HAIR,FACEHAIR,LIPS,NO_UNDERWEAR)
+	inherent_traits = list(TRAIT_RESISTLOWPRESSURE,TRAIT_RESISTHIGHPRESSURE,TRAIT_NOBREATH)//temporary nobreath
```
Comment (Jujumatic) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm:10]: no high pressure on lavaland but this can be removed once nobreath is fixed too

File: hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 2]
```diff
@@ -57,7 +59,7 @@
 			if(0 to 30) // too low, harmful
 				H.adjustBruteLoss(10)
```
Comment (Jujumatic) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm:60]: ow, that's going to trigger every time the tick goes down. hope that doesn't tick too often...

**Comment by Toxici11i** [comment]
-1
removes both golems and ashwalkers meaning that dwarves have no competition

HGTOW anyone?

Also its not headmin approved just because you made it. stop being gay

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 3]
```diff
@@ -1,117 +1,48 @@
...
-		H.underwear = "Nude"
-		H.update_body()
-
-/obj/effect/mob_spawn/human/ash_walker/Initialize(mapload, datum/team/ashwalkers/ashteam)
+	uses = -1
+	banType = "lavaland"
+	short_desc = "You're a dwarf through and through. You sustain yourself off alcohol and the crops you grow."
+	flavour_text = "The last thing you remember before blacking out was seeing an Elven army attacking your fortress.\
+	You prepared to make a final stand against the filthy tree huggers when suddenly their wizards performed some kind \
+	of blasphemous elf magic. In the fortress you were..."
```
Comment (Jujumatic) [inline code/game/objects/structures/ghost_role_spawners.dm:18]: filthy ELVES

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 4]
```diff
@@ -1,117 +1,48 @@
...
-
-/obj/effect/mob_spawn/human/exile/Destroy()
-	new/obj/structure/fluff/empty_sleeper(get_turf(src))
-	return ..()
-
-/obj/effect/mob_spawn/human/exile/special(mob/living/new_spawn)
-	new_spawn.fully_replace_character_name(null,"Wish Granter's Victim ([rand(1,999)])")
-	var/wish = rand(1,4)
-	switch(wish)
+	var/dwarfstory = rand(1,2)
```
Comment (Jujumatic) [inline code/game/objects/structures/ghost_role_spawners.dm:23]: this should be pretty easy to expand and make more random stories as dwarves develop, nice

**Comment by ghost** [comment]
its cool +1

**Comment by AnthonyJNgo** [comment]
@Toxici11i I'm not approving my own idea, the idea has been greenlit by former and present Headmins. 
#10769 by KBB, #12365 which was the pre-pr to this one. 
Dwarves will NEVER have competition from ash lizards or golems because neither of those ghost roles have ANY content. Ash lizard is spawn and try to bum rush station, golem is spawn and try to bum rush station. Neither of them have actual gameplay or features. Dwarves is different and will have actual things that you can do such as smithing ores that you mine, crafting, fighting new npcs such as goblin villages, etc.

File: hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 5]
```diff
@@ -15,6 +16,7 @@
 	offset_features = list(OFFSET_UNIFORM = list(0,0), OFFSET_ID = list(0,0), OFFSET_GLOVES = list(0,0), OFFSET_GLASSES = list(0,0), OFFSET_EARS = list(0,0), OFFSET_SHOES = list(0,0), OFFSET_S_STORE = list(0,0), OFFSET_FACEMASK = list(0,-3), OFFSET_HEAD = list(0,-3), OFFSET_HAIR = list(0,-4), OFFSET_FACE = list(0,-3), OFFSET_BELT = list(0,0), OFFSET_BACK = list(0,0), OFFSET_SUIT = list(0,0), OFFSET_NECK = list(0,-3))
 	mutantlungs = /obj/item/organ/lungs/dwarven
 	mutanttongue = /obj/item/organ/tongue/dwarven
+	coldmod = 0
```
Comment (Jujumatic) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm:19]: their beards and alcohol will protect them against the snow

File: code/datums/atmosphere/planetary.dm [hunk 6]
```diff
@@ -3,21 +3,14 @@
 	id = LAVALAND_DEFAULT_ATMOS
 
 	base_gases = list(
-		/datum/gas/oxygen=5,
-		/datum/gas/nitrogen=10,
+		/datum/gas/carbon_dioxide=100,
```
Comment (Jujumatic) [inline code/datums/atmosphere/planetary.dm:6]: this may cause normal people without o2 canisters to instantly pass out and potentially die

File: hippiestation/code/modules/surgery/organs/lungs.dm [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /obj/item/organ/lungs/dwarven
     name = "dwarven lungs"
-    desc = "A pair of quite small lungs. They look different than normal human's ones."
+    desc = "A pair of extremely small lungs. They don't look human at all." //this description was bad
```
Comment (Jujumatic) [inline hippiestation/code/modules/surgery/organs/lungs.dm:3]: maybe you could say they're soaked in alcohol or something idk

**Review by Jujumatic — APPROVED** [review]
looks good so far though the lavaland atmos will probably need to be fixed pretty quickly after this goes in, public beta time go

**Comment by AnthonyJNgo** [comment]
Travis having a literal stroke because the maps won't compile. Of course they won't compile, I smashed the delete key on them.

**Comment by wheezl** [comment]
+1

**Comment by TheSpartyn** [comment]
At least keep the syndicate base, the comms agent role is a very unique one that dwarves wont replace.

**Comment by ghost** [comment]
yeah agreed with spartyn no need to remove so much content. syndicate base is ok

**Comment by Pyko1** [comment]
-1 not really a big fan of removing those templates from the actual code, why not just make them not be a part of the list of possible ruins?

No harm keeping them there, if we ever want to use them again.

**Comment by GuyonBroadway** [comment]
Yeah why are we nixing all of the lavaland roles anyway? Sounds like trying to sneak in grudgecode on top of features.

**Comment by ghost** [comment]
@GuyonBroadway cuz dwarves are BETTER

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 8]
```diff
@@ -1,117 +1,48 @@
...
 
-//Preserved terrarium/seed vault: Spawns in seed vault structures in lavaland. Ghosts become plantpeople and are advised to begin growing plants in the room near them.
-/obj/effect/mob_spawn/human/seed_vault
-	name = "preserved terrarium"
-	desc = "An ancient machine that seems to be used for storing plant matter. The glass is obstructed by a mat of vines."
-	mob_name = "a lifebringer"
-	icon = 'icons/obj/lavaland/spawners.dmi'
-	icon_state = "terrarium"
-	density = TRUE
+//Dwarves, the ultimate ghost role.
```
Comment (ghost) [inline code/game/objects/structures/ghost_role_spawners.dm:3]: couldn't you have just unticked this file, and made a new one?

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by Maroonenahmoron** [comment]
-1, I like the enthusiasm but not the removal of other spawners. They’re isolated places for people to learn new techniques for chemistry, mining, botany and surviving in lavaland.

Placeholders for dwarves look good though, how about making their fortress have some wooden flooring? I would love to design a starting base.

**Comment by GuyonBroadway** [comment]
@steamp0rt it's a modular system jackass, no reason we can't have everything.

**Comment by AnthonyJNgo** [comment]
Comms agent imo is legitimately terrible.

@guyonbroadway I was told by KBB that Dwarves should be the role that replaces all other lavaland roles since we’ll also be changing lavaland. 

@Maroon they’ll still be able to experiment with those things, and learn to survive in lavaland.

File: code/game/objects/structures/ghost_role_spawners.dm (hunk 8 above)
Comment (AnthonyJNgo) [reply code/game/objects/structures/ghost_role_spawners.dm:3]: I could have, yes, but I think we’re going a bit overboard with “stash this away for later”. We want to make the code better, not have a million unused things thrown around in our codebase. This is a direct replacement.

Comment (Pyko1) [reply code/game/objects/structures/ghost_role_spawners.dm]: Codebase size is hardly an issue, and at least personally i believe its always better to keep the code even if the feature is never used in the game again. Someone could still use the code as an example or recycle parts of it.

**Comment by Pyko1** [comment]
Yeah even if dwarves are going to be the defacto lavaland race, the other templates should be saved, as if anything else they could be loaded for some gimmic.

Not to mention in future if we run dwarfs 100% of the time, people might get bored and want to see them return once in a while, so then it would be an easy change in configs, instead of having to re-add them.

**Comment by GuyonBroadway** [comment]
Look me in the eye and say a three way war between assliggers, Lavaland Gaylems and Dorfs isn't the best thing that could happen to lavaland.

**Comment by KingTerrytheTerrible** [comment]
we dont have the pop for that and you aren't helping our pop issue by leaving

**Comment by xTrainx** [comment]
Don’t kill lavaland sub people that’s gay

**Comment by Manly-Dwarf** [comment]
stop voting on this it's already approved
we have said for a long ass time that the other spawners would be removed once dwarves would come in, and only now you noticed?

1: A lot of the spawners take up a huge amount of space and offer legitimately nothing to the round (see: beachbums, who when spawned only murder each other before committing suicide). Or it's the comms agent, who will proceed to lag the server by blowing up the base and metagrudge his killer by stalking him on cameras. Or attempt to read WGW.
The ash lizards are the closest thing to something that has effort, but they're really simple. If Dwarves are going to be as complicated as they are, there's no reason to have these other spawners.
2: The spawners are not maintained and have nothing added to them. For a long time everyone has agreed that they're forgettable. Again, Beach Bums. Now we have a problem with them being removed?
3: There's nothing you can do as any of the spawners that you won't be able to do as a Dwarf. They just take up a ton of space.
4: If the argument is that Lavaland clutter doesn't matter because it's huge, you're forgetting that the Dwarves are going to have actual basebuilding in them and likely aren't going to want to have to deal with being surrounded by massive, useless ghost role ruins.

The most valuable ghost role by far is the syndicate base, and probably the only one I'd consider keeping since its isolated on a lake any way. 80% of rounds it either goes silent or just gets blown up by someone who picked the ghost role while bored, assuming they didn't pick comms agent to start screeching who their killer was over comms.
I do agree it's probably not worthwhile to outright remove the templates from the code.

**Comment by Manly-Dwarf** [comment]
like who even picks the veterinarian ghost role
are you seriously telling me "noooo don't remove that" when it does nothing, and Dwarves will already be able to have that neat base
i hope you guys are arguing for the lavalizards/syndi outpost and not that

File: hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm (hunk 2 above)
Comment (Manly-Dwarf) [reply hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm:60]: its only temporary as far as we've concerned, but I've told vivadel a thousand times that it's absolutely atrocious

File: code/game/objects/structures/ghost_role_spawners.dm (hunk 3 above)
Comment (Manly-Dwarf) [reply code/game/objects/structures/ghost_role_spawners.dm:18]: finna rewrite this probably

File: code/datums/atmosphere/planetary.dm (hunk 6 above)
Comment (Manly-Dwarf) [reply code/datums/atmosphere/planetary.dm:6]: its a good thing miners get a lot of those at least

File: hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm (hunk 5 above)
Comment (Manly-Dwarf) [reply hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm:19]: join the dwarfcord to find out why

**Comment by AnthonyJNgo** [comment]
I’m going to make some changes for the dwarf PR. I’ll comment out all the other lavaland roles, I’ll leave the syndicate base, ash lizards, golems in rotation.

File: hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm (hunk 2 above)
Comment (AnthonyJNgo) [reply hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm:60]: I think it’ll go over a lot smoother during actual gameplay, if it doesn’t it’s literally a one line change so no big deal.

File: code/game/objects/structures/ghost_role_spawners.dm (hunk 3 above)
Comment (AnthonyJNgo) [reply code/game/objects/structures/ghost_role_spawners.dm:18]: If you have a better intro thing absolutely, I just made it up on the spot to explain why/how a random patch of snowy mountain terrain landed on lavaland. Once the whole lavaland is converted to snow bio it’ll be different of course.

File: code/game/objects/structures/ghost_role_spawners.dm (hunk 4 above)
Comment (AnthonyJNgo) [reply code/game/objects/structures/ghost_role_spawners.dm:23]: Yep. The nice thing about this is the role is decided at roundstart so you can’t just suicide repeatedly to get a different one.

**Comment by GuyonBroadway** [comment]
Just cus it's been approved in discord doesn't mean it can;t be subject to change following criticism, going "NUH UH DUN DEAL!" is oranges level reasoning.

**Comment by Manly-Dwarf** [comment]
that's ok, i explained my reasoning here and agree that there's a valid point to be made about the syndi lavaland base. we could probably keep that, replace the comms agent with a botany room, and be good to go. or we could move it to space since there is (or was? dunno if vivadel also removed space roles) a comms agent in space as well as one on lavaland.
comms agent is the real problem with the lavaland base. otherwise it's pretty much fine.

all of the others are bad. golems are only used by nerds like toxic to put a million of them on the station and then run around murderboning since free golems have no orders, so i am not surprised he would vouch for them.

**Comment by GuyonBroadway** [comment]
Also we should not delete or disable them outright. Just make them not viable for the lavaland spawner.

**Comment by Manly-Dwarf** [comment]
agreed

**Comment by GuyonBroadway** [comment]
Who needs nations when you have dwarves vs Ashliggers vs Golems?

**Comment by Manly-Dwarf** [comment]
dwarves and humans are already going to be the main competing force. i don't think you understand this.
dwarves aren't just able to kill the miners, they are also directly competing with them for the exact same resource. the more they stick around, the more they'll be allowed to thrive and become more and more annoying for the miners. there's a good chance there will be rounds where dwarves or miners don't run into each other at all, and also a good chance that the miners will be wiped out while the dwarves get CO2 tanks and hijack their shuttle. sometimes a miner will rush the dwarves and try to destroy them immediately, and i am currently thinking of ways to prevent this
they don't just exist as a silly lavaland role, they also exist to make mining more interesting and more dangerous. most miners by now already know how to deal with the lacking variety in mobs, and dwarves also gives us an excuse to add more of those.

it might be cool to have a respawn dome where people can 1v1, but we already had that in VR in the past. it might be better to just make a thunderdome not operated entirely by admins and make a system where ghosts can fight each other. i have wanted to do that for a while but i am poopy at code

**Comment by Toxici11i** [comment]
>>>not implying dwarves will just build 9 spears and become unga spearthrower

**Comment by Manly-Dwarf** [comment]
free shrug

**Comment by cris5311** [comment]
-1 dwarfs good removing lavaland roles bad

**Comment by Manly-Dwarf** [comment]
i am genuinely very surprised by how many people, despite me telling them for a year on end that dwarves were going to replace all ghost roles, are now upset that dwarves are doing exactly what they were supposed to do
what is up with you people

**Comment by cris5311** [comment]
I mean throw out the vet but the other lavaland roles have uses

**Comment by Manly-Dwarf** [comment]
yeah we should find a way to merge the botany base and syndie base, and keep those one way or another. completely agree there, those have use.

**Comment by mkikt4743** [comment]
dwarfs good +1

**Comment by GuyonBroadway** [comment]
In a perfect world we could have a system whereby the main "off station mining hazard" is picked randomly or determined by config.

**Comment by FantasticFwoosh** [comment]
In a perfect world we could finish the Dwarf Ruin + elves/goblin/funstuff bells and whistles side activities that come with and then make the entire thing cost the **_entire ruin budget_** so that we don't have to go full :b: :registered: over removing a ton of content.

**Comment by cris5311** [comment]
Also ashliggers make dwarfs more interesting. i mean who dosnt like hunting liggers?

**Comment by Manly-Dwarf** [comment]
"a ton of content"
did i miss something? barely even a month ago i could talk to everyone about how little and how worthless the ghost role ruins were and how they added almost nothing, and everyone agreed
now suddenly everyone's in a hissy over a proper merging/replacement under the name "dwarves"
also no, ashlizards don't make dwarves more interesting. they really just kind of detract. if we had a higher population, maybe they'd be worthwhile. but what would they do that the dwarves couldn't?
let's keep the syndicate outpost, add botany to it and then we have that perfect world. if you just wanna practice silly stuff, you still can. if you want to compete against the miners, you can (as a dwarf). there's no faults with this way to go as I see it.

i still like that ghost thunderdome idea too

**Comment by Maroonenahmoron** [comment]
Dwarves good, syndicate base is good... whatever happened to syndicate cargo shuttle? If they could order supplies for “syndicate experiments” then it should be fine to get rid of plant people+golems.

Liggers could be a rare ruin so they don’t take away dwarf glory but make it like an event where they can either interact or war with each other.

The other spawners such as beach bum + veterinarians will be overshadowed so they’re good to be booted out.

**Comment by Fikou** [comment]
just make the golem ship have a botany room and a chemistry/virology room and theres no need for podmen or syndies lol

**Comment by TheTrigg** [comment]
I've always liked how golems can setup their own RND stuff on their ship and build almost everything from there.
On an especionally long round, I once managed to build an entire working Xenobiology setup.
Golems are cool.

And the Liberator's legacy is a pretty unique item. If the golems really are to be removed, at least move that awesome little box to some other ruin.

**Comment by Manly-Dwarf** [comment]
gonna be completely honest, might just take the existing code and try to go at another sprint
trying to merge dwarves in this incomplete state without all of this stuff is makin me realize it would just be another ghost ruin. not much is in there yet, and while you can say "yet", that don't change that it's fuckin barren right now and there's no reason to remove the other roles

**Comment by Manly-Dwarf** [comment]
tl;dr it's WAY too early to merge them, not enough content to justify removals. -1

**Comment by Fikou** [comment]
@AnthonyJNgo i can pr you a bigger golem ship map for when you finish this to replace podmen and lavaland syndies if you want

**Comment by mkikt4743** [comment]
it doesn't fit in with rest of lavaland with the snow and shit, make lavaland snowland first

**Comment by AnthonyJNgo** [comment]
Update, still actively working on this.

**Comment by AnthonyJNgo** [comment]
Promise I'll have some more commits out soon, been busy upkeeping other aspects of the server.

**Comment by FantasticFwoosh** [comment]
Looking kind of stale at the moment, anything you can update us on?

**Comment by AnthonyJNgo** [comment]
Sorry, I have been pretty busy working on preparing everything for the host swap. Host swap SHOULD be finalized maybe later today and then I’ll be able to get more work on this done. Thanks for the patience.

**Comment by SireTurret** [comment]
so sad this is nowhere near completion

**Comment by AnthonyJNgo** [comment]
Yeetus deletus

**Comment by karmaisblackandbluepilled** [comment]
so sad this is nowhere near completion

**Comment by AnthonyJNgo** [comment]
This PR will be resubmitted as separate smaller PRs to ensure what we want gets in and what we don't want stays out or gets modified.
